HTTP Server.core/IFunctionality.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace HTTPServer.core
+{
+    public interface IFunctionality
+    {
+        byte[] Execute(Request request);
+        bool ShouldRun(Request request, IPathContents pathContents);
```

Comment:
I needed something that RequestHandler was not dependent on to know which one to run.  I originally used strings as keys, but the logic needed to be more complex.